### ISSUE TITLE
Add verbosity argument

### ISSUE THREAD

**Issue by FrickTobias** [fillipe-gsm/python-tsp#19]
When using `logger` in a parent script and importing `python-tsp`,  `python-tsp` will print its logging messages every iteration to the parent scripts logger, with no off-switch. This crowds your terminal and significantly reduces performance. 

# Solution suggestion
Add a verbosity option
```
def function(verbose: bool=False):
    if verbose:
        logger.info(...)
```
# Current workaround
My current workaround it to have a logger level specifically for every function in `python-tsp` which has info statements
```
logging.getLogger("python_tsp.heuristics.local_search").setLevel(logging.WARNING)
```
# Speed testing 

With logging prints
```
5.8179931640625
5.103294849395752
5.7765209674835205
5.681611061096191
5.754772186279297
6.6095380783081055
6.0969202518463135
5.9088897705078125
```

Without logging prints
```
2.521873950958252
2.845388889312744
2.74272084236145
2.5899457931518555
3.1590828895568848
1.9179699420928955
2.808474063873291
2.76981782913208
2.2262699604034424
2.3329758644104004
```

## Full code
```{bash}
for i in {1..10}; do 
    python speedtest.py --seed $i --with-logger >> speed-test-with-prints.txt
done
```
```{bash}
for i in {1..10}; do 
    python speedtest.py --seed $i >> speed-test-without-prints.txt
done
```
```{python}
import numpy as np
import python_tsp
import time
import logging
from argparse import ArgumentParser
from python_tsp.heuristics import solve_tsp_local_search
from python_tsp.distances import euclidean_distance_matrix

# args
parser = ArgumentParser()
parser.add_argument("--seed", required=True, type=int)
parser.add_argument("--with-logger", action="store_true")
args = parser.parse_args()

# Test if logging makes a difference
if args.with_logger:
    logging.basicConfig(level=logging.INFO)

# Generate random points
np.random.seed(args.seed)
coords = np.random.randint(0, 1500, (150, 2))

# calculate dist matrix
distance_matrix = euclidean_distance_matrix(coords)

# measure time it takes to solve
t1 = time.time()
permutation, distance = solve_tsp_local_search(distance_matrix)
t2 = time.time()

print(t2 - t1)
```

**Comment by fillipe-gsm** [comment]
Hi, @FrickTobias , thanks for pointing that out and for the script, it helped a log with debugging.
I was using a File Handler and was sure that it would only log to a file if the `log_file` variable were provided, but that was not correct :sorry:.

I researched quite a while and in summary I found that using `logging` is apparently too complex for this simple function; we get a lot of different behaviors if the root level configures logging differently. I played with `propagate = True` and other things but this is simply too risky to use.

I think it is safer to revert back to the usual `print`. It will be less invasive and it will still be possible to keep the same API and expected behaviors.

I will try to create a PR quickly and upgrade the lib to version 0.4.1.

**Comment by FrickTobias** [comment]
@fillipe-gsm Thanks for getting and back and for providing a nice tool.

Sounds reasonable. I wasn't sure if you wanted to maintain the log file writing capability, but if not I threw together a [fork](https://github.com/FrickTobias/python-tsp) where I changed the following: 

`local_search.py`/`simulated_annealing.py`
 - Replaced `logger.info()` with `print()` 
 - Replaced `logger.warning()` with `print("WARNING: ")`
 - Removed `log_file` test

`test_local_search.py`/`test_simulated_annealing.py`
 - Changed `capLog` to `io.StringIO()` to test if early stop warning is being printed to stdout
 - Removed `log_file` argument

**Comment by fillipe-gsm** [comment]
> @fillipe-gsm Thanks for getting and back and for providing a nice tool.

Thank you for using it!

> but if not I threw together a [fork](https://github.com/FrickTobias/python-tsp) where I changed the following

That's a very good solution. Would you mind opening a pull request?

The only small modifications I would like to make are:
- Move the imports
```python
from io import StringIO
import sys
```
to the top since they belong to the standard library;

- Keep the `log_file` functionality: it may not be a good idea to remove variables and add a breaking change if someone is already using it.

Of course, if you'd like you can open the pull request as it is and I can make the other minor changes in other commits. In any case, it would be very nice to add your contribution to this small project.

**Comment by FrickTobias** [comment]
Yeah, that is probably a good idea. Breaking stuff can be dangerous.

I updated it with the imports moved and logger statements added again. I went with pretty much the easiest solution possible for keeping the logging functionality but it should work. Feel free to edit it if you want something a bit more neat.

I opened a [pull request](#20).

```
if log_file:
    logger.info(msg)
print(msg)
```

**Comment by fillipe-gsm** [comment]
Thanks for opening the PR!
I added some suggestions. If you need I am happy to help.

Of course, If you think you might be busy I can add the suggestions in more commits over there, no problem at all.

**Comment by FrickTobias** [comment]
This was fixed in PR #20

**Comment by fillipe-gsm** [comment]
For future reference, I updated the project to version 0.3.1, so the issue described here should be fixed.

Thanks @FrickTobias for the help and discussion; I added you as a contributor in the README :1st_place_medal:
